### PR TITLE
[Feature] Replace old auth api with new privilege api

### PR DESCRIPTION
**Pull request by nshangyiming**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

Going to replace the use of api in old Auth module in new privilege framework, this is the first PR, since the situation where we need to handle the repalcement is complicated, we will split this action into 4-5 PRs.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java [hunk 1]
```diff
@@ -1105,10 +1105,12 @@ private void getAlterJobV2Infos(Database db, List<AlterJobV2> alterJobsV2,
             if (alterJob.getDbId() != db.getId()) {
                 continue;
             }
-            if (ctx != null) {
-                if (!GlobalStateMgr.getCurrentState().getAuth()
-                        .checkTblPriv(ctx, db.getFullName(), alterJob.getTableName(), PrivPredicate.ALTER)) {
-                    continue;
+            if (!GlobalStateMgr.getCurrentState().isUsingNewPrivilege()) {
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java:1108]: If I'm not mistaken, when `isUsingNewPrivilege` is true, all privilege checks will pass. Is it what you want?

Comment (nshangyiming) [reply fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java:1108]: Yes, this method is only used by `show /proc` statement, and discussed with PM @wangsimo0 before, the final decision is that we won't check privilege for `show /proc` statement.

**Review by gengjun-git — DISMISSED** [review]

**Comment by nshangyiming** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 51 / 84 (60.71%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/ExportMgr.java|0|13|00.00%|[221, 223, 224, 226, 227, 228, 230, 231, 232, 234, 235, 236, 238]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzerUtils.java|0|5|00.00%|[138, 139, 140, 146, 147]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|0|5|00.00%|[1317, 1319, 1584, 1586, 3069]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|2|5|40.00%|[1385, 1387, 1389]|
|:large_blue_circle:|com/starrocks/load/DeleteHandler.java|3|4|75.00%|[652]|
|:large_blue_circle:|com/starrocks/alter/SchemaChangeHandler.java|4|5|80.00%|[1112]|
|:large_blue_circle:|com/starrocks/alter/MaterializedViewHandler.java|4|5|80.00%|[1014]|
|:large_blue_circle:|com/starrocks/privilege/PrivilegeManager.java|26|30|86.67%|[745, 746, 788, 789]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeCheckerV2.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/privilege/PrivilegeCollection.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/privilege/DefaultAuthorizationProvider.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 54 / 89 (60.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzerUtils.java|0|5|00.00%|[138, 139, 140, 146, 147]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|0|5|00.00%|[1317, 1319, 1584, 1586, 3069]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|2|5|40.00%|[1385, 1387, 1389]|
|:large_blue_circle:|com/starrocks/privilege/PrivilegeManager.java|27|46|58.70%|[605, 617, 628, 639, 651, 663, 678, 690, 703, 717, 733, 746, 748, 766, 804, 805, 806, 849, 866]|
|:large_blue_circle:|com/starrocks/load/DeleteHandler.java|3|4|75.00%|[652]|
|:large_blue_circle:|com/starrocks/alter/SchemaChangeHandler.java|4|5|80.00%|[1112]|
|:large_blue_circle:|com/starrocks/alter/MaterializedViewHandler.java|4|5|80.00%|[1014]|
|:large_blue_circle:|com/starrocks/load/ExportMgr.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeCheckerV2.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/privilege/PrivilegeCollection.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/privilege/DefaultAuthorizationProvider.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|1|1|100.00%|[]|

**Review by gengjun-git — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**